### PR TITLE
Use Gradle 7 and increase source compat to 11

### PR DESCRIPTION
**Pull request by fawind**
## After this PR

==COMMIT_MSG==
Use Gradle 7
==COMMIT_MSG==

### CODE REVIEW

File: gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/tasks/LazyCreateStartScriptTask.java [hunk 1]
```diff
@@ -37,12 +37,14 @@ public LazyCreateStartScriptTask() {
         return mainClassName;
     }
 
+    @SuppressWarnings("deprecation")
```
Comment (CRogers) [inline gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/tasks/LazyCreateStartScriptTask.java:40]: Is something in here going to be deprecated in Gradle 8?

File: gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/ConfigTarTask.java [hunk 2]
```diff
@@ -38,9 +37,10 @@ public final AbstractCopyTask from(Object... sourcePaths) {
     }
 
     @Override
-    @SuppressWarnings("RawTypes")
+    @SuppressWarnings({"RawTypes", "deprecation"})
     public final AbstractCopyTask from(Object sourcePath, Closure closure) {
-        return this.from(sourcePath, new ClosureBackedAction<>(closure));
+        // TODO(fwindheuser): Replace usage of 'ClosureBackedAction' before moving to Gradle 8.
```
Comment (CRogers) [inline gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/ConfigTarTask.java:42]: Me, adding this line literally last week

side_eye_monkey_small

File: gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java [hunk 3]
```diff
@@ -174,7 +178,7 @@ private void ensureLockfileIsUpToDate(List<ProductDependency> productDeps) {
                         "%s does not exist, please run `./gradlew %s --write-locks` and commit the resultant file",
                         relativePath, getName()));
             } else {
-                String fromDisk = GFileUtils.readFile(lockfile);
```
Comment (CRogers) [inline gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java:177]: This method is actually a footgun anyway, it uses the default charset...

File: gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java [hunk 4]
```diff
@@ -174,7 +178,7 @@ private void ensureLockfileIsUpToDate(List<ProductDependency> productDeps) {
...
+                String fromDisk = new String(Files.readAllBytes(lockfile.toPath()), StandardCharsets.UTF_8);
```
Comment (CRogers) [inline gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java]: If we are using java 11 in this repo yet, we can do `Files.readString`, which also helpfully uses UTF-8 by default.

File: gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java [hunk 5]
```diff
@@ -186,10 +190,10 @@ private void ensureLockfileIsUpToDate(List<ProductDependency> productDeps) {
     }
 
     /** Provide a rich diff so the user understands what change will be made before they run --write-locks. */
-    private Optional<String> diff(File existing, String upToDateContents) {
+    private Optional<String> diff(File existing, String upToDateContents) throws IOException {
         try {
             File tempFile = Files.createTempFile("product-dependencies", "lock").toFile();
-            GFileUtils.writeFile(upToDateContents, tempFile);
+            Files.write(tempFile.toPath(), upToDateContents.getBytes(StandardCharsets.UTF_8));
```
Comment (CRogers) [inline gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java]: Similarly `Files.writeString`.

File: gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/ProductDependencyIntrospectionPlugin.java [hunk 6]
```diff
@@ -103,8 +101,13 @@ private static String getMinimumProductVersion(Project project, String group, St
             return Optional.empty();
         }
 
-        return Optional.of(ProductDependencyLockFile.fromString(
-                GFileUtils.readFile(lockFile), project.getVersion().toString()));
+        try {
+            String rawLockFile = new String(Files.readAllBytes(lockFile.toPath()), StandardCharsets.UTF_8);
```
Comment (robert3005) [inline gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/ProductDependencyIntrospectionPlugin.java]: surprised intellij didn't suggest `Files#readString`

Comment (fawind) [reply gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/ProductDependencyIntrospectionPlugin.java]: Unfortunately, we are still on source compat 1.8 and I did not want to increase it as part of this pr :(

Currently checking if this is still required or if we can raise it to 11.

File: gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/tasks/LazyCreateStartScriptTask.java (hunk 1 above)
Comment (fawind) [reply gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/tasks/LazyCreateStartScriptTask.java:40]: The `setMainClassName` on the parent class is deprecated in favor of setting the value through the property.
I think it's the easiest to also remove this method once the parent method gets removed in Gradle 8 (?).

File: gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/ProductDependencyIntrospectionPlugin.java (hunk 6 above)
Comment (fawind) [reply gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/ProductDependencyIntrospectionPlugin.java]: We are good to use source compat 11 now!

File: gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java (hunk 5 above)
Comment (fawind) [reply gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java]: Updated

File: gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java (hunk 4 above)
Comment (fawind) [reply gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java]: Updated

**Comment by CRogers** [comment]
:+1:

**Pull request merged**